mlflow/transformers.py

```diff
@@ -1320,28 +1320,8 @@ class _TransformersWrapper:
     def __init__(self, pipeline, inference_config=None):
         self.pipeline = pipeline
         self.inference_config = inference_config
-        if inference_config:
-            self._validate_inference_config_keys(pipeline, inference_config)
         self._conversation = None
-
-    @staticmethod
-    def _validate_inference_config_keys(pipeline, inference_config):
-        """
-        Raise an Exception if invalid inference_config keys for the given Pipeline and
-        Model were provided during save.
-        """
-        invalid_keys = []
-        for key in inference_config:
-            if not hasattr(pipeline, key) and not hasattr(pipeline.model, key):
-                invalid_keys.append(key)
-        if invalid_keys:
-            raise MlflowException(
-                "The Inference Configuration overrides that were saved with "
-                "this model are not compatible as overrides to either the "
-                "Pipeline or the Model. Please re-save the model with valid "
-                f"inference configuration keys. Invalid keys: {invalid_keys}",
-                error_code=INVALID_PARAMETER_VALUE,
-            )
+        self._supported_custom_generator_types = {"InstructionTextGenerationPipeline"}
```

Comment:
This is Dolly.

Reply:
Can we add a comment and say this is Dolly? A link to the code also helps.

Reply:
`InstructionTextGenerationPipeline` is our own custom class name and `transformers` doesn't have it, right?

https://github.com/databrickslabs/dolly/blob/0eadcb7b0648d496d67243a7d572b413560be661/training/generate.py#L65

Reply:
yep, it's a custom subclass of TextGenerationPipeline. I modified the logic to do an exact match of the as-saved pipeline instance type so that we don't have inadvertent manipulation of any other TextGenerationPipeline types

Reply:
@harupy I added an NB note with a link to the Hub page (that contains the model card info) with a caveat to say that all variants are covered in this pipeline type parsing logic